language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/DamlGenMap.java

```diff
@@ -13,12 +13,24 @@
 
 public final class DamlGenMap extends Value {
 
-    private static DamlGenMap EMPTY = new DamlGenMap(Collections.EMPTY_MAP);
+    private final Map<Value, Value> map;
 
-    private final java.util.Map<Value, Value> value;
+    /**
+     * The map that is passed to this constructor must not be changed
+     * once passed.
+     */
+    private DamlGenMap(@NonNull Map<@NonNull Value, @NonNull Value> value) {
+        this.map = value;
+    }
+
+    private static @Nonnull DamlGenMap fromPrivateMap(@NonNull Map<@NonNull Value, @NonNull Value> map){
```

Comment:
Did you mean to use both `Nonnull` and `NonNull`?

Reply:
fixed